Source/SwiftLintFramework/Rules/UntypedErrorRule.swift

```diff
@@ -0,0 +1,81 @@
+        "(?:"   + // Start of the first non-capturing group
+        "\\s+"  + // At least one any type of whitespace character
+        "(?:"   + // Start of the alternative non-capturing group
+        "let"   + // `let` keyword
+        "|"     + // OR
+        "var"   + // `var` keyword
+        ")"     + // End of the alternative non-capturing group
+        "\\s+"  + // At least one any type of whitespace character
+        "\\w+"  + // At least one any type of word character
+        ")"     + // End of the first non-capturing group
+        "(?:"   + // Start of the second non-capturing group
+        "\\s*"  + // Zero or unlimited any whitespace character
+        ")"     + // End of the second non-capturing group
+        "\\{"     // Start scope character
+    )
+
+    public static let description = RuleDescription(
+        identifier: "untyped_error",
+        name: "Untyped error rule",
+        description: "Catch statements should not declare error variables without type casting",
+        kind: .lint,
+        nonTriggeringExamples: [
+            "do {\n try foo() \n} catch {\n}",
+            "do {\n try foo() \n} catch Error.invalidOperation {\n} catch {\n}",
+            "do {\n try foo() \n} catch let error as? MyError {\n} catch {\n}",
+            "do {\n try foo() \n} catch var error as? MyError {\n} catch {\n}"
+        ],
+        triggeringExamples: [
+            "do {\n try foo() \n} ↓catch var error {\n}",
+            "do {\n try foo() \n} ↓catch let error {\n}",
+            "do {\n try foo() \n} ↓catch let someError {\n}",
+            "do {\n try foo() \n} ↓catch var someError {\n}",
+            "do {\n try foo() \n} \n ↓catch let e {\n}"
+        ])
+
+    public func validate(file: File,
+                         kind: StatementKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+
+        guard kind == .brace,
+            let offset = dictionary.offset else {
+                return []
+        }
+
+        return file.lines
+            .lazy
+            .filter { $0.byteRange.contains(offset) }
+            .reduce([]) { _, line in
+                let range = NSRange(location: 0, length: line.content.bridge().length)
+                return type(of: self).regularExpression.matches(in: line.content, options: [], range: range)
+                    .map { match -> StyleViolation in
+                        let characterOffset = line.range.lowerBound + match.range.lowerBound
+                        let location = Location(file: file,
+                                                characterOffset: characterOffset)
+                        return StyleViolation(ruleDescription: type(of: self).description,
+                                              severity: configuration.severity,
+                                              location: location,
+                                              reason: configuration.consoleDescription)
```

Comment:
Is there a reason why you're iterating through lines?

I have the impression that it'd be faster to make this a regular rule (not `ASTRule`) and just match the regex on the whole file, validating some syntax kinds (e.g. that `catch` is a `keyword`, as well as `let`/`var`).

Reply:
@marcelofabri Thanks for the review! This suggestion was a really spot on. I have decided to use ASTRule to be able to solve the issue with the commented out code. However the syntax kind validation works great, simplified the code by tons and should be faster too.